index.js

```diff
@@ -407,107 +407,137 @@ function buildInnerObject (context, location) {
+    const allOfSchemaLocation = allOfSchemasLocation.getPropertyLocation(i)
+    locations.push(allOfSchemaLocation)
+  }
 
-    if (allOfSchema.format !== undefined) {
-      if (
-        mergedSchema.format !== undefined &&
-        mergedSchema.format !== allOfSchema.format
-      ) {
-        throw new Error('allOf schemas have different format values')
-      }
-      mergedSchema.format = allOfSchema.format
-    }
+  const mergedLocation = mergeSchemas(context, locations)
+  return mergedLocation
+}
 
-    if (allOfSchema.nullable !== undefined) {
-      if (
-        mergedSchema.nullable !== undefined &&
-        mergedSchema.nullable !== allOfSchema.nullable
-      ) {
-        throw new Error('allOf schemas have different nullable values')
-      }
-      mergedSchema.nullable = allOfSchema.nullable
-    }
+function mergeSchemas (context, locations) {
+  const mergedSchema = {}
 
-    if (allOfSchema.properties !== undefined) {
-      if (mergedSchema.properties === undefined) {
-        mergedSchema.properties = {}
-      }
-      Object.assign(mergedSchema.properties, allOfSchema.properties)
-    }
+  for (let location of locations) {
+    let schema = location.schema
+    resolveRelativeRefs(schema, location.schemaId)
 
-    if (allOfSchema.additionalProperties !== undefined) {
-      if (mergedSchema.additionalProperties === undefined) {
-        mergedSchema.additionalProperties = {}
-      }
-      Object.assign(mergedSchema.additionalProperties, allOfSchema.additionalProperties)
-    }
+    if (schema.$ref) {
+      const { $ref, ...schemaWithoutRef } = schema
+      const refSchemaLocation = resolveRef(context, location, $ref)
+      refSchemaLocation.schema = clone(refSchemaLocation.schema)
 
-    if (allOfSchema.patternProperties !== undefined) {
-      if (mergedSchema.patternProperties === undefined) {
-        mergedSchema.patternProperties = {}
-      }
-      Object.assign(mergedSchema.patternProperties, allOfSchema.patternProperties)
-    }
+      const newSchemaLocation = new Location(
+        schemaWithoutRef,
+        location.schemaId,
+        location.jsonPointer
+      )
 
-    if (allOfSchema.required !== undefined) {
-      if (mergedSchema.required === undefined) {
-        mergedSchema.required = []
-      }
-      mergedSchema.required.push(...allOfSchema.required)
+      location = mergeSchemas(
+        context,
+        [refSchemaLocation, newSchemaLocation]
+      )
+      schema = location.schema
     }
 
-    if (allOfSchema.oneOf !== undefined) {
-      if (mergedSchema.oneOf === undefined) {
-        mergedSchema.oneOf = []
-      }
-      mergedSchema.oneOf.push(...allOfSchema.oneOf)
-    }
+    for (const key in schema) {
+      const value = schema[key]
 
-    if (allOfSchema.anyOf !== undefined) {
-      if (mergedSchema.anyOf === undefined) {
-        mergedSchema.anyOf = []
+      if (key === '$id') continue
+      if (key === 'allOf') {
```

Comment:
`switch` *might* be faster here because of all the string comparisons

Reply:
Ditto